apps/linux/vim.talon

```diff
@@ -0,0 +1,799 @@
+    # "u" gets inserted into INSERT mode
+    user.vim_mode_change_timeout = 0.3
+
+###
+# Actions - Talon generic_editor.talon implementation
+###
+#
+# NOTE: You can disable generic_editor.talon by renaming it, and still fully
+# control vim. These are more for people that are used to the official talon
+# editor commands that want to trial vim a bit. I don't personally use most of
+# the actions here, so they have not been thoroughly tested
+#
+###
+action(edit.find):
+    user.vim_normal_mode_exterm_key("/")
+action(edit.find_next):
+    user.vim_normal_mode_key("n")
+action(edit.word_left):
+    user.vim_normal_mode_key("b")
+action(edit.word_right):
+    user.vim_normal_mode_key("w")
+action(edit.left):
+    key(left)
+action(edit.right):
+    key(right)
+action(edit.up):
+    key(up)
+action(edit.down):
+    key(down)
+action(edit.line_start):
+    user.vim_normal_mode_key("^")
+action(edit.line_end):
+    user.vim_normal_mode_key("$")
+action(edit.file_end):
+    user.vim_normal_mode_key(G)
+action(edit.file_start):
+    user.vim_normal_mode("gg")
+action(edit.page_down):
+    user.vim_normal_mode_exterm_key("ctrl-f")
+action(edit.page_up):
+    user.vim_normal_mode_exterm_key("ctrl-b")
+
+action(edit.extend_line_end):
+    user.vim_visual_mode("$")
+action(edit.extend_left):
+    user.vim_visual_mode("h")
+action(edit.extend_right):
+    user.vim_visual_mode("l")
+action(edit.extend_line_up):
+    user.vim_visual_mode("k")
+action(edit.extend_line_down):
+    user.vim_visual_mode("j")
+action(edit.extend_word_left):
+    user.vim_visual_mode("b")
+action(edit.extend_word_right):
+    user.vim_visual_mode("w")
+action(edit.extend_line_start):
+    user.vim_visual_mode("^")
+action(edit.extend_file_start):
+    user.vim_visual_mode("gg")
+action(edit.extend_file_end):
+    user.vim_visual_mode("G")
+
+action(edit.indent_more):
+    user.vim_normal_mode(">>")
+action(edit.indent_less):
+    user.vim_normal_mode("<<")
+action(edit.delete_line):
+    user.vim_normal_mode("dd")
+action(edit.delete):
+    user.vim_normal_mode_key(x)
+
+# note these are for mouse highlighted copy/paste. shouldn't be used for actual
+# vim commands
+action(edit.copy):
+    key(ctrl-shift-c)
+action(edit.paste):
+    key(ctrl-shift-v)
+
+###
+# `code/vim.py` actions based on vimspeak
+###
+# commands that can be triggered in visual or normal mode, and generally don't
+# have counting, etc
+<user.vim_normal_counted_motion_command>:
+    insert("{vim_normal_counted_motion_command}")
+#<user.vim_counted_command>:
+#    insert("{vim_counted_command}")
+<user.vim_motions_all_adjust>:
+    insert("{vim_motions_all_adjust}")
+# XXX - needs to be adjusted to handle the actions associated with key presses
+<user.vim_normal_counted_action>:
+    insert("{vim_normal_counted_action}")
+
+###
+# File editing and management
+###
+# NOTE: using `save` alone conflicts too much with the `say`
+save file:
+    user.vim_command_mode(":w\n")
```

Comment:
Instead of creating a new 'save file' or 'save all' command, I'd suggest maybe implementing the existing actions
https://github.com/knausj85/knausj_talon/blob/87efebdcae0ad23db7a29dee899b7177d8391e40/apps/win/edit.talon#L157

and sticking with the global commands where possible?
https://github.com/knausj85/knausj_talon/blob/87efebdcae0ad23db7a29dee899b7177d8391e40/misc/standard.talon#L61
?

Reply:
ya that works

Reply:
should note that `save all` is fundamentally different though, in that `save file` saves the current buffer, and `save all` saves all open buffers. so will still end up having a `file save all` or something that isn't an `edit.talon` action probably

Reply:
Make sense